client/rpc/src/eth/block.rs

```diff
@@ -239,34 +204,36 @@ where
 			)));
 		}
 
-		let id = match frontier_backend_client::native_block_id::<B, C>(
-			self.client.as_ref(),
-			self.backend.as_ref(),
-			Some(number),
-		)
-		.await?
-		{
-			Some(id) => id,
-			None => return Ok(None),
-		};
-		let substrate_hash = self
-			.client
-			.expect_block_hash_from_id(&id)
-			.map_err(|_| internal_err(format!("Expect block number from id: {}", id)))?;
-		let schema = fc_storage::onchain_storage_schema(self.client.as_ref(), substrate_hash);
-		let block = self
-			.overrides
-			.schemas
-			.get(&schema)
-			.unwrap_or(&self.overrides.fallback)
-			.current_block(substrate_hash);
-
-		match block {
+		let block_info = self.block_info_by_number(number).await?;
+		match block_info.block {
 			Some(block) => Ok(Some(U256::from(block.transactions.len()))),
 			None => Ok(None),
 		}
 	}
 
+	pub async fn block_transaction_receipts(
+		&self,
+		number: BlockNumber,
+	) -> RpcResult<Option<Vec<Receipt>>> {
+		let block_info = self.block_info_by_number(number).await?;
+		let Some(statuses) = block_info.clone().statuses else {
+			return Ok(None);
+		};
+
+		let mut receipts = Vec::new();
+		let transactions: Vec<(H256, usize)> = statuses
+			.iter()
+			.map(|tx| (tx.transaction_hash, tx.transaction_index as usize))
+			.collect();
+		for (hash, index) in transactions {
```

Comment:
Don't we already have the `receipts` in `block_info.receipts` here?

I mean, the data is there, the issue would be build the JSON `Receipt` objects from those `ethereum::ReceiptV3`, right? Maybe implementing a `TryFrom` for it?

BTW, pretty useful `BlockInfo` impl =)